execution-engine/src/wasi/strace.rs

```diff
@@ -9,30 +9,99 @@
 //! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
 //! information on licensing and copyright.
 
-#![allow(unused_variables)] // while most of the functions are unimplemented
-
 use super::common::MemoryHandler;
 use crate::fs::FileSystemResult;
 use std::fmt;
 
-pub struct Strace {}
+enum TraceState {
```

Comment:
Can you also add `rustdoc`-style comments to this enum and its fields explaining what they are?

Reply:
Done.